src/components/TimeBasedLineChart.js

```diff
@@ -0,0 +1,753 @@
+    });
+  }
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.height !== null || newProps.width !== null || newProps.margin !== null) {
+      const height = newProps.height || this.props.height;
+      const width = newProps.width || this.props.width;
+      const margin = newProps.margin || this.props.margin;
+
+      const adjustedWidth = width - margin.right - margin.left;
+      const adjustedHeight = height - margin.top - margin.bottom;
+
+      this.setState({
+        adjustedHeight,
+        adjustedWidth
+      });
+    }
+  }
+
+  _getAreaBelowZero (data) {
+    const area = d3.svg.area()
+      .x(d => {
+        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix()
+
+        return this._getXScaleValue(currentDate);
+      })
+      .y0(this.state.adjustedHeight - 10)
+      .y1(this._getYScaleValue(0));
+
+    return area(data);
+  }
+
+  _getFlatLine (data) {
+    const flatLine = d3.svg.line()
+      .x(d => {
+        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix()
+
+        return this._getXScaleValue(currentDate);
+      })
+      .y(this.state.adjustedHeight);
+
+    return flatLine(data);
+  }
+
+  _getLine (data) {
+    const line = d3.svg.line()
+      .x(d => {
+        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix()
+
+        return this._getXScaleValue(currentDate);
+      })
+      .y(d => {
+        return this._getYScaleValue(d.value);
+      });
+
+    return line(data);
+  }
+
+  _getSliceMiddle (timeStamp) {
+    return Math.floor(this._getSliceWidth(timeStamp) / 2);
+  }
+
+  _getSliceWidth (timeStamp) {
+    const rangeType = this.props.rangeType;
+    const startOf = timeStamp ? moment.unix(timeStamp).startOf(rangeType).unix() : moment(timeStamp).startOf(rangeType).unix();
+    const endOf = timeStamp ? moment.unix(timeStamp).endOf(rangeType).unix() : moment(timeStamp).endOf(rangeType).unix();
+
+    return Math.floor(this._getXScaleValue(endOf) - this._getXScaleValue(startOf));
+  }
+
+  _getSplitData () {
+    const future = [];
+    const past = [];
+
+    this.props.data.forEach(item => {
+      const currentDate = moment.unix(item.timeStamp).startOf(this.props.rangeType);
+
+      if (currentDate.isBefore(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        past.push(item);
+      }
+
+      if (currentDate.isAfter(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        future.push(item);
+      }
+
+      if (currentDate.isSame(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        past.push(item);
+        future.push(item);
+      }
+    });
+
+    return [past, future];
+  }
+
+  _getYAxisTicks (data, numTicks = 4) {
+    const maxValue = d3.max(this.props.data, d => {
+      const value = d.above ? d.value + d.above : d.value + 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
```

Comment:
Rounding to the nearest 1000s so that the ticks end up being 0, 1k, 2k, 3k, etc....